_overviews/scala3-book/scala-for-python-devs.md

```diff
@@ -0,0 +1,1041 @@
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+Scala has other specialized `Map` classes for different needs.  
+
+
+### Sets
+
+The Python set is like the mutable Scala `Set` class.
+
+<table cellspacing="1" cellpadding="2" border="1">
+  <tbody>
+
+    <tr>
+      <th valign="top">Python</th>
+      <th valign="top">Scala</th>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Creation</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>set = {"a", "b", "c"}</code></td>
+      <td valign="top"><code>val set = Set(1,2,3)</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Duplicate elements</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>set = {1,2,1}
+        <br># set: {1,2}</code></td>
+      <td valign="top"><code>val set = Set(1,2,1)
+        <br>// set: Set(1,2)</code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+Scala has other specialized `Set` classes for different needs.
+
+
+### Tuples 
+
+The Python and Scala tuples are also similar:
+
+<table cellspacing="1" cellpadding="2" border="1">
+  <tbody>
+    <tr>
+      <th valign="top">Python</th>
+      <th valign="top">Scala</th>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Creation</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>t = (11, 11.0, "Eleven")</code></td>
+      <td valign="top"><code>val t = (11, 11.0, "Eleven")</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Access elements</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>t[0]&nbsp;&nbsp; # 11
+        <br>t[1]&nbsp;&nbsp; # 11.0</code>
+      </td>
+      <td valign="top"><code>t(0)&nbsp;&nbsp; // 11
+        <br>t(1)&nbsp;&nbsp; // 11.0</code>
+      </td>
+    </tr>
+  </tbody>
+</table>
+
+
+
+## Methods on collections classes
+
+Python and Scala have several of the same common functional methods available to them:
+
+- `map`
+- `filter`
+- `reduce`  
+
+If you’re used to using these methods with lambda expressions in Python, you’ll see that Scala has a similar approach with methods on its collections classes. To demonstrate this functionality, here are two sample lists:  
+
+```scala
+list = (1,2,3)           // python
+val list = List(1,2,3)   // scala
+```
+
+Those lists are used in the following table, that shows how to apply mapping and filtering algorithms to it:
+
+<table cellspacing="1" cellpadding="2" border="1">
+  <tbody>
+    <tr>
+      <th valign="top">Python</th>
+      <th valign="top">Scala</th>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Mapping with a comprehension</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>x = [i*10 for i in list]</code></td>
+      <td valign="top"><code>val x = for i &lt;- list yield i * 10</code></td>
+    </tr>
+
+    <tr>
+      <td colspan="2" class="table-desc-row">Filtering with a comprehension</td>
+    </tr>
+    <tr>
+      <td valign="top"><code>evens = [i for i in list if i % 2 == 0]</code></td>
```

Comment:
On the website those two are not really separated.  Maybe we should have some kind of a vertical line between the two columns?

Reply:
We have a bit of a problem with horizontal alignment, since the page only has a fixed width. I propose we show the languages one after another. Vertical lines often are visually problematic.

Reply:
Yes, @b-studios created a little prototype to show what the code would look like with the languages one after another. Changing the format of these pages is on my to-do list.

(FWIW, when I first created the pages this way I didn’t know if the format of the website was going to change, and I preferred the side-by-side approach if we had more width to work with.)